### PR TITLE
Add support for PlaceholderAPI

### PR DESCRIPTION
**Pull request by SeruhioX**
Added compatibility with placeholderAPI

Example:

```
/svs add <if> placeholder:%luckperms_has_permission_somePermission%=yes
/svs add <msg> OK
/svs add <endif>
```

////

```
/svs add <if> placeholder:%player_name%=Seruhio|Peter|NickName
/svs add <msg> OK
/svs add <endif>
```

### CODE REVIEW

**Comment by EpiCanard** [comment]
I think you can reformat all your changes you have a mix of tab and spaces.
The format of other files is 4 spaces.

**Comment by amnrmain** [comment]
> I think you can reformat all your changes you have a mix of tab and spaces.
> The format of other files is 4 spaces.

Sure :)

**Comment by SeruhioX** [comment]
> I think you can reformat all your changes you have a mix of tab and spaces.
> The format of other files is 4 spaces.

Sure :)

**Comment by EpiCanard** [comment]
Good work, thank you :)

**Comment by czymstef** [comment]
This sounds like a nice addition, thanks for the PR. It would need some additions to the external documentation tho.
I'll leave this open for now while I discuss further steps with Exloki.

**Comment by EpiCanard** [comment]
Any news ?

**Comment by EpiCanard** [comment]
@CalibeR50  Can we at least merge this Pull Request please ? This feature opens many possibilities that we sadly can not use for the moment.

Thanks in advance for you answer

**Comment by SeruhioX** [comment]
I think this feature is essential for this plugin, as it adds infinite possibilities.

I was waiting a long time for the author to add it and asked several times... When the code was published I opened this request since someone had to do it.

We will continue waiting for an answer.

Greetings

**Comment by czymstef** [comment]
Hi, 

I think, as long as nobody actively maintains the project, pull requests for features are not really important, especially if those involve additional third party dependencies. The features would neither be documented nor released, which is why I don't currently want to accept the PR.
If you need the features yourself, you can easily build svs with the respective features on your local machine.

If one is willing to take over maintaining the project, I'd be happy to provide the necessary resources to get this going again.

I hope you can understand my reasoning here.

Greetings

**Comment by EpiCanard** [comment]
Hello

If you agree, I would like to continue to actively maintains the project. This plugin must continue to live.
I will make sure that the project continues to work properly, update the documentation regularly and answer questions from the players.

Thanks in advance for your answer

Best regards

**Comment by czymstef** [comment]
Hi, 

sounds great. Can you leave me a message at svs@czymm.de with your discord? I guess we can discuss anything further there.

Greetings

File: src/main/java/de/czymm/serversigns/hooks/HookManager.java [hunk 1]
```diff
@@ -48,5 +50,10 @@ public void tryInstantiateHooks(boolean deepVerbose) {
             ServerSignsPlugin.log("Unable to load Vault dependency - certain economy and permission features will be disabled");
             ServerSignsPlugin.log("Please download Vault at http://dev.bukkit.org/server-mods/vault/ for Economy and \"Permission grant\" support.");
         }
+        try {
+            placeholderAPI.instantiateHook();
+        } catch (Exception ex) {
+            ServerSignsPlugin.log("Unable to load PlaceholderAPI dependency - placeholder features will be disabled");
```
Comment (EpiCanard) [inline src/main/java/de/czymm/serversigns/hooks/HookManager.java:56]: Can you add a link to PlaceholderAPI download ? https://www.spigotmc.org/resources/placeholderapi.6245/

File: src/main/java/de/czymm/serversigns/hooks/PlaceholderAPIHook.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.bukkit.Bukkit;
+
+public class PlaceholderAPIHook {
+
+    protected ServerSignsPlugin pl;
+
+    public PlaceholderAPIHook(ServerSignsPlugin plugin) throws Exception {
+        pl = plugin;
+
+        if (Bukkit.getPluginManager().getPlugin("PlaceholderAPI") == null) {
```
Comment (EpiCanard) [inline src/main/java/de/czymm/serversigns/hooks/PlaceholderAPIHook.java:31]: Can you add a condition to check if PlaceholderAPI is disabled ?

File: src/main/java/de/czymm/serversigns/parsing/operators/PlaceholderOperator.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+public class PlaceholderOperator extends ConditionalOperator {
+
+    public PlaceholderOperator() {
+        super("placeholder", true);
+    }
+
+    @Override
+    public ParameterValidityResponse checkParameterValidity(String params) {
+        boolean isValid = params.indexOf('=') > 0 && params.length() - 2 >= params.indexOf('=');
+        return new ParameterValidityResponse(isValid, "Parameter must be in the format <placeholder>=<match>");
```
Comment (EpiCanard) [inline src/main/java/de/czymm/serversigns/parsing/operators/PlaceholderOperator.java:18]: Can you add support of > and < for number only ?
That we can support comparison between number ?

**Review by EpiCanard — CHANGES_REQUESTED** [review]
You forgot dependency to PlaceholderAPI in pom.xml, can you add it ?

File: src/main/java/de/czymm/serversigns/parsing/operators/PlaceholderOperator.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package de.czymm.serversigns.parsing.operators;
```
Comment (EpiCanard) [inline src/main/java/de/czymm/serversigns/parsing/operators/PlaceholderOperator.java:1]: Can you add GNU Licence header please ?

**Review by EpiCanard — CHANGES_REQUESTED** [review]

**Comment by Segysmundo** [comment]
Hey there, SeruhioX appears to be abroad with his job and won't be able to take care of this any time soon.

Do you think you would be able to implement this on your own?

**Comment by EpiCanard** [comment]
I can do it but I don't have much free time right now to apply the requested changes.

**Comment by EpiCanard** [comment]
I merge to **feature/placeholderApi** to continue myself the work.

**Pull request merged**